stock_warehouse_flow/models/stock_warehouse_flow.py

```diff
@@ -0,0 +1,367 @@
+                self.pack_type_id.default_location_src_id = self.pack_stock_loc_id
+                self.pack_type_id.default_location_dest_id = self.output_stock_loc_id
+        # Enable relevant locations (pack/output) based on the delivery steps
+        self.warehouse_id._update_location_delivery(self.delivery_steps)
+        try:
+            with self.env.cr.savepoint():
+                wh_vals = {
+                    "delivery_steps": self.delivery_steps,
+                    "out_type_id": self.out_type_id.id,
+                }
+                if self.pick_type_id:
+                    wh_vals.update(
+                        {
+                            "pick_type_id": self.pick_type_id.id,
+                            "wh_output_stock_loc_id": self.output_stock_loc_id.id,
+                        }
+                    )
+                if self.pack_type_id:
+                    wh_vals.update(
+                        {
+                            "pack_type_id": self.pack_type_id.id,
+                            "wh_pack_stock_loc_id": self.pack_stock_loc_id.id,
+                        }
+                    )
+                wh_vals["out_type_id"] = self.out_type_id.id
+                self.warehouse_id.write(wh_vals)
+                vals = self.warehouse_id.delivery_route_id.copy_data()[0]
+                raise Exception()
+        except Exception:
+            pass
+        vals.update(
+            {
+                "name": f"{self.warehouse_id.name}: {self.name}",
+                "active": True,
+                "flow_id": self.id,
+                # Give the priority to routes/rules not tied to a flow
+                # when the initial move is created by the procurement
+                "sequence": 1000,
+            }
+        )
+        self.delivery_route_id = self.env["stock.location.route"].create(vals)
+        self.to_picking_type_id = self.out_type_id
+        # Subscribe the route to the warehouse
+        if self.delivery_route_id.warehouse_selectable:
+            self.warehouse_id.route_ids |= self.delivery_route_id
+
+    def action_generate_route(self):
+        for flow in self:
+            flow._generate_delivery_route()
+        return True
+
+    def _search_for_move_domain(self, move):
+        return [
+            ("from_picking_type_id", "=", move.picking_type_id.id),
+            ("carrier_ids", "in", move.group_id.carrier_id.ids),
+            ("delivery_route_id", "!=", False),
+        ]
+
+    def _is_valid_for_move(self, move):
+        self.ensure_one()
+        if not self.move_domain:
+            return self
+        domain = safe_eval(self.move_domain or "[]")
+        if not domain:
+            return self
+        return move.filtered_domain(domain)
+
+    @api.model
+    def _search_for_move(self, move):
+        """Return the first flow matching the move."""
+        domain = self._search_for_move_domain(move)
```

Comment:
Wondering: could you make this work w/ a list of moves? They could be grouped by `(picking_type_id, carrier_id)` after search. No?

Reply:
Not sure how to do that, I prefer to keep it like this right now. Let's see if it becomes an issue later on...